pkg/apparmor/apparmor_linux_template.go

```diff
@@ -21,6 +21,10 @@ profile {{.Name}} flags=(attach_disconnected,mediate_deleted) {
   # Allow signals from privileged profiles and from within the same profile
   signal (receive) peer=unconfined,
   signal (send,receive) peer={{.Name}},
+  # Allow certain signals from OCI runtimes (podman, runc and crun)
+  signal (receive) peer={/usr/bin/,/usr/sbin/,}runc,
+  signal (receive) peer={/usr/bin/,/usr/sbin/,}crun*,
+  signal (receive) set=(int, quit, kill, term) peer={/usr/bin/,/usr/sbin/,}podman,
```

Comment:
podman uses kill(pid,0) to check if the process is still alive, would that get blocked by this list?
Is there any practical reason to limit podman to certain signals?

Reply:
The permission to check a pid is given in the imported `abstractions/base` along with the permissions to self-signal  and receive unconfined processes

```
  # Allow unconfined processes to send us signals by default
  signal (receive) peer=unconfined,

  # Allow us to signal ourselves
  signal peer=@{profile_name},

  # Checking for PID existence is quite common so add it by default for now
  signal (receive, send) set=("exists"),
```

The two lines that already appear in the linux_template are really superfluous as they are handled by `abstractions/base` and have been since apparmor 2.9 which introduced signal filtering a decade or more ago.

The reason for filtering signals is the same reason for denying access to anything via apparmor - defence against error and compromise. SIGSTOP to `conmon` for example. The `conmon` code appears to use default dispositions for most signals. Arguably SIGQUIT could be removed from the list to avoid a Core default disposition (does anything allow core dumps any more?).

I couldn't tell from testing (or the podman code) what signals podman expects to send. I just got a DENIED error in the logs. I plumped for a standard set and that seemed to do the trick. 

```
May 14 11:14:41 srv-omzr6 kernel: audit: type=1400 audit(1715685281.392:118): apparmor="DENIED" operation="signal" class="signal" profile="containers-default-0.57.4" pid=7458 comm="conmon" requested_mask="receive" denied_mask="receive" signal=term peer="podman"
```

If you have the actual signals `podman` intends to send, then we can limit to that.

Reply:
> The reason for filtering signals is the same reason for denying access to anything via apparmor - defence against error and compromise. SIGSTOP to conmon for example. The conmon code appears to use default dispositions for most signals. Arguably SIGQUIT could be removed from the list to avoid a Core default disposition (does anything allow core dumps any more?).

Sure but I don't see how anyone can expect a container to be safe in case of a podman compromise. Blocking certain signals for security seems pointless. As a user you can run `podman kill` to send any signal, sure that is funnelled through the oci runtime (crun,runc) and you have to allow everything there anyway. Because nothing can prevent a compromised from calling the oci runtime preventing podman from sending a certain signal does not add any security IMO.

I don't think we send any other signals today but I cannot be sure about that. But if it changes in the future the profiles will not allow it causing unnecessary bugs. And no maintainer will remember to edit the apparmor profile until a bug will be reported.

Reply:
It's the usual debate about whether to be open to all (the standard unix approach) or be locked down (which is what selinux and apparmor add in). 

I can't really help with that. It's a philosophical point for the podman project to decide. I don't know in detail what conmon is there to do and whether it requires signal filtering. 

All I want is for my database containers to stop failing.

Reply:
Yeah fair enough, the current profile doesn't work at all and this fixes the problems we are seeing today so I am ok merging it as is. If we start sending other signals we fail again but it is unlikely enough that I can live with such possibilities.